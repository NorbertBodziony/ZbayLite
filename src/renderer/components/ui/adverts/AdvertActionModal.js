import React from 'react'
import PropTypes from 'prop-types'
import * as R from 'ramda'
import { AutoSizer } from 'react-virtualized'
import { Scrollbars } from 'react-custom-scrollbars'

import Grid from '@material-ui/core/Grid'
import Button from '@material-ui/core/Button'
import Typography from '@material-ui/core/Typography'
import { withStyles } from '@material-ui/core/styles'

import Modal from '../Modal'

const styles = theme => ({
  window: {
    width: 570,
    position: 'absolute',
    top: '50%',
    left: '50%',
    transform: 'translate(-50%, -50%)'
  },
  main: {
    marginTop: 32,
    padding: '0px 32px'
  },
  backgroundImage: {
    width: '100%',
    height: 225,
    backgroundColor: 'yellow',
    borderRadius: 4
  },
  tagContainer: {
    width: 156,
    height: 46,
    backgroundColor: 'rgba(0, 0, 0, 0.6)',
    borderRadius: 4
  },
  tag: {
    color: theme.palette.colors.white,
    fontSize: 28,
    lineHeight: '34px',
    fontWeight: 500
  },
  hash: {
    opacity: 0.6
  },
  title: {
    marginTop: 16,
    fontWeight: 500,
    color: theme.palette.colors.titleGray
  },
  descriptionContainer: {
    marginTop: 10
  },
  description: {
    letterSpacing: 0.4,
    color: theme.palette.colors.darkGray
  },
  priceUsd: {
    marginLeft: 9,
    color: theme.palette.colors.trueBlack,
    fontSize: 13,
    lineHeight: '24px',
    letterSpacing: 0.4
  },
  priceZcash: {
    marginLeft: 3,
    color: theme.palette.colors.darkGray,
    fontSize: 13,
    lineHeight: '24px',
    letterSpacing: 0.4
  },
  priceContainer: {
    marginTop: 10,
    padding: '0px 20px'
  },
  actionsContainer: {
    marginTop: 10,
    padding: '0px 20px'
  },
  button: {
    textTransform: 'none',
    marginRight: 12,
    minWidth: 126,
    height: 31,
    padding: 0,
    margin: 0,
    backgroundColor: theme.palette.colors.buttonGray
  },
  buttonString: {
    fontStyle: 'normal',
    fontWeight: 'normal',
    fontSize: 14,
    lineHeight: '10px',
    color: 'rgba(0, 0, 0, 0.35)'
  }
})

export const AdvertActionModal = ({
  classes,
  open,
  payload,
  handleBuy,
  handleClose,
  handleMessage
}) => (
  payload ? <Modal
    open={open}
    handleClose={handleClose}
    classes={{ window: classes.window }}
    title={`#${payload.tag} @${payload.offerOwner}`}
    fullPage
  >
    <AutoSizer>
      {({ width, height }) => (
        <Scrollbars autoHideTimeout={500} style={{ width: width, height: height }}>
          <Grid container dirention={'column'} justify={'center'} alignContent={'flex-start'} className={classes.main}>
            <Grid container className={classes.backgroundImage} style={{ background: `url(${payload.background})` }} item justify={'center'} alignContent={'center'}>
              <Grid container item className={classes.tagContainer} style={{ width: 30 + payload.tag.length * 15 }} justify={'center'} alignItems={'center'}>
                <Typography variant={'h1'} className={classes.tag}><span className={classes.hash}>#</span>{payload.tag}</Typography>
              </Grid>
            </Grid>
            <Grid container item>
              <Typography variant={'body2'} className={classes.title}>{payload.title}</Typography>
            </Grid>
            <Grid container className={classes.descriptionContainer} item>
              <Typography variant={'caption'} className={classes.description}>{payload.description}</Typography>
            </Grid>
          </Grid>
          <Grid container direction={'row'} className={classes.priceContainer} wrap={'nowrap'} item justify={'flex-start'}>
            <Grid item>
              <Typography variant={'caption'} className={classes.priceUsd}>${payload.priceUSD}</Typography>
            </Grid>
            <Grid item>
              <Typography variant={'caption'} className={classes.priceZcash}>({`${payload.priceZcash} ZEC`})</Typography>
            </Grid>
          </Grid>
          <Grid container direction={'row'} className={classes.actionsContainer} wrap={'nowrap'} item justify={'flex-start'}>
            <Grid item>
              <Button onClick={handleMessage} className={classes.button}><span className={classes.buttonString}>Message Seller</span></Button>
            </Grid>
            <Grid item>
              <Button onClick={handleBuy} className={classes.button}><span className={classes.buttonString}>Buy Now</span></Button>
            </Grid>
          </Grid>
        </Scrollbars>
      )}
    </AutoSizer>
  </Modal>
    : null
)

AdvertActionModal.propTypes = {
  classes: PropTypes.object.isRequired,
  open: PropTypes.bool.isRequired,
  payload: PropTypes.shape({
    tag: PropTypes.string.isRequired,
    priceUSD: PropTypes.string.isRequired,
    priceZcash: PropTypes.string.isRequired,
    offerOwner: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
  }),
  handleBuy: PropTypes.func.isRequired,
  handleMessage: PropTypes.func.isRequired,
  handleClose: PropTypes.func.isRequired
}

AdvertActionModal.defaultProps = {
  payload: {
    tag: '',
    priceUSD: '',
    priceZcash: '',
    offerOwner: '',
    description: '',
    title: ''
  }
}

export default R.compose(
  React.memo,
  withStyles(styles)
)(AdvertActionModal)
