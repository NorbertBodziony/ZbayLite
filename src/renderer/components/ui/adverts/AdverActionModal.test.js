import React from 'react'
import { shallow } from 'enzyme'

import { AdvertActionModal } from './AdvertActionModal'
import { mockClasses } from '../../../../shared/testing/mocks'

describe('SendMessagePopover', () => {
  it('renders popover', () => {
    const payload = {
      tag: 'dirtyBike',
      description: 'Great quality bike for half the price as a name brand dirt bike! The X4',
      title: 'Apollo X4 110cc Dirt Bike for...',
      priceUSD: '300',
      offerOwner: 'roks33',
      priceZcash: '4000',
      background: '98c9e4113d76a80d654096c9938fb1a3.svg'
    }
    const result = shallow(
      <AdvertActionModal
        open
        handleBuy={jest.fn()}
        handleMessage={jest.fn()}
        handleClose={jest.fn()}
        payload={payload}
        classes={mockClasses}
      />
    )
    expect(result).toMatchSnapshot()
  })
})
