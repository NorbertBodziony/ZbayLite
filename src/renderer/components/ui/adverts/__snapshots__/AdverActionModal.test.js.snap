// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SendMessagePopover renders popover 1`] = `
<Memo(WithStyles(Modal))
  classes={
    Object {
      "window": "window-mock",
    }
  }
  fullPage={true}
  handleClose={[MockFunction]}
  open={true}
  title="#dirtyBike @roks33"
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    direction="column"
  >
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      className="user-mock"
      container={true}
      direction="row"
      item={true}
      justify="space-between"
    >
      <WithStyles(ForwardRef(Grid))
        alignItems="center"
        container={true}
        direction="row"
        item={true}
        xs={true}
      >
        <WithStyles(ForwardRef(Grid))
          className="alignAvatar-mock"
          item={true}
        >
          <Jdenticon
            size="50"
            value="roks33"
          />
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          className="userInfoDiv-mock"
          item={true}
        >
          <WithStyles(ForwardRef(Grid))
            container={true}
            direction="column"
          >
            <WithStyles(ForwardRef(Grid))
              item={true}
            >
              <WithStyles(ForwardRef(Typography))
                className="username-mock"
                variant="h4"
              >
                roks33
              </WithStyles(ForwardRef(Typography))>
            </WithStyles(ForwardRef(Grid))>
            <WithStyles(ForwardRef(Grid))
              item={true}
            >
              <WithStyles(ForwardRef(Typography))
                className="userInfo-mock"
                variant="caption"
              >
                Seller
              </WithStyles(ForwardRef(Typography))>
            </WithStyles(ForwardRef(Grid))>
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <WithStyles(ForwardRef(Button))
          className="buttonMessage-mock"
          onClick={[Function]}
          variant="outlined"
        >
          Message
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={true}
    >
      <AutoSizer
        disableHeight={false}
        disableWidth={false}
        onResize={[Function]}
        style={Object {}}
      >
        <Component />
      </AutoSizer>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      className="bottomInfo-mock"
      item={true}
    >
      <WithStyles(ForwardRef(Grid))
        alignItems="center"
        container={true}
        justify="space-between"
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={true}
        >
          <WithStyles(ForwardRef(Typography))
            className="bottomHashtag-mock"
            variant="body1"
          >
            #
            dirtyBike
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Grid))
            className="priceContainer-mock"
            container={true}
            direction="row"
            item={true}
            justify="flex-start"
            wrap="nowrap"
          >
            <WithStyles(ForwardRef(Grid))
              item={true}
            >
              <WithStyles(ForwardRef(Typography))
                className="priceUsd-mock"
                variant="caption"
              >
                $
                300
              </WithStyles(ForwardRef(Typography))>
            </WithStyles(ForwardRef(Grid))>
            <WithStyles(ForwardRef(Grid))
              item={true}
            >
              <WithStyles(ForwardRef(Typography))
                className="priceZcash-mock"
                variant="caption"
              >
                (
                4000 ZEC
                )
              </WithStyles(ForwardRef(Typography))>
            </WithStyles(ForwardRef(Grid))>
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <WithStyles(ForwardRef(Button))
            className="buyButton-mock"
            onClick={[Function]}
            variant="contained"
          >
            Buy
          </WithStyles(ForwardRef(Button))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</Memo(WithStyles(Modal))>
`;
